core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexTemplateService.java

```diff
@@ -163,10 +154,62 @@ public void onFailure(String source, Throwable t) {
             }
 
             @Override
-            public ClusterState execute(ClusterState currentState) {
+            public ClusterState execute(ClusterState currentState) throws Exception {
                 if (request.create && currentState.metaData().templates().containsKey(request.name)) {
                     throw new IndexTemplateAlreadyExistsException(request.name);
                 }
+
+                boolean indexCreated = false;
+                String dummyIndexPrefix = ".dummy_";
+                try {
+
+                    //create index service for parsing and validating mappings
+                    Settings.Builder settingsBuider = Settings.settingsBuilder()
+                        .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
+                        .put(request.settings);
+
+                    if (request.settings.get(IndexMetaData.SETTING_NUMBER_OF_SHARDS) == null) {
+                        settingsBuider.put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1);
+                    }
+
+                    if (request.settings.get(IndexMetaData.SETTING_NUMBER_OF_REPLICAS) == null) {
+                        settingsBuider.put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0);
+                    }
+
+                    Settings dummySettings = settingsBuider.build();
+
+                    if (indicesService.hasIndex(dummyIndexPrefix + request.name)) {
+                        indicesService.removeIndex(dummyIndexPrefix + request.name, "created for parsing template mapping");
```

Comment:
I don't think we should delete the index at the beginning of the request. Instead, we should throw an error if the special index name exists. That way, if it is somehow an index that was created by the user, we would let them know, and not silently delete their data.